### DISCUSSION TITLE
Updating a Non-Catalog Request

### DISCUSSION THREAD

**Discussion by DrMarkX** [Snow-Shell/servicenow-powershell#228] [Q&A] [answered]
Is there a way to update non-catalog requests? I've tried...
`Get-ServiceNowRecord -Table 'u_non_catalog_request' -ID NON0292380 | Update-ServiceNowRecord -Values @{work_notes = "Updated by PowerShell"}`
But that returns the error message: _Invoke-WebRequest : The remote server returned an error: (400) Bad Request._

**Comment by gdbarron** [comment]
Are you sure it's the Update command that's failing and not Get?  Also, sometimes the class of a table, which is what is picked up from the pipeline, is not the same as the database name.  Try adding `-Table 'u_non_catalog_request'` to the Update command.

**Reply by DrMarkX** [reply]
I ran that as two separate lines and as one line.  Both gave the same error - the error showed when I ran
`Update-ServiceNowRecord -Table 'u_non_catalog_request' -ID NON0292380 -Values @{work_notes = "Updated by PowerShell"}`

**Comment by gdbarron** [comment]
This is a custom table so any number of issues might be occurring.  What results do you get if you run `Get-ServiceNowRecord -Table 'u_non_catalog_request' -ID 'NON292380' -Property 'sys_class_name', 'sys_id', 'number'`?  If empty, can you get 1 record from that table and provide the results so I can see all the properties?

**Reply by DrMarkX** [reply]
I had to pull data from another request to get those data.
```
sys_id                           number     sys_class_name     
------                           ------     --------------
9a55c8e8875c2190b7f3468acebb354a NON0301378 Non Catalog Request
```

**Comment by DrMarkX** [comment, marked as answer]
This is the solution.
`Get-ServiceNowRecord -Table 'u_non_catalog_request' -ID NON0301378 | Update-ServiceNowRecord -Table 'u_non_catalog_request' -Values @{"comments" = "This is a new additional comment.";"notify" = $true;"type" = "Additional Comments";"created_by" = $SNOWCred.UserName}`
